app/src/main/java/com/example/pantrypal/addTagsPage.java

```diff
@@ -0,0 +1,130 @@
+        CheckBox savoryTag = findViewById(R.id.savoryTag);
+        CheckBox saltyTag = findViewById(R.id.saltyTag);
+        CheckBox bitterTag = findViewById(R.id.bitterTag);
+        CheckBox umamiTag = findViewById(R.id.umamiTag);
+
+        CheckBox drinkTag = findViewById(R.id.drinkTag);
+        CheckBox snackTag = findViewById(R.id.snackTag);
+        CheckBox breakfastTag = findViewById(R.id.breakfastTag);
+        CheckBox lunchTag = findViewById(R.id.lunchTag);
+        CheckBox dinnerTag = findViewById(R.id.dinnerTag);
+        CheckBox dessertTag = findViewById(R.id.dessertTag);
+
+        CheckBox beefTag = findViewById(R.id.beefTag);
+        CheckBox porkTag = findViewById(R.id.porkTag);
+        CheckBox chickenTag = findViewById(R.id.chickenTag);
+        CheckBox turkeyTag = findViewById(R.id.turkeyTag);
+        CheckBox duckTag = findViewById(R.id.duckTag);
+        CheckBox poultryTag = findViewById(R.id.poultryTag);
+        CheckBox fishTag = findViewById(R.id.fishTag);
+        CheckBox lambTag = findViewById(R.id.lambTag);
+        CheckBox rabbitTag = findViewById(R.id.rabbitTag);
+        CheckBox shellfishTag = findViewById(R.id.shellfishTag);
+        CheckBox vegetarianTag = findViewById(R.id.vegetarianTag);
+        CheckBox veganTag = findViewById(R.id.veganTag);
+
+        checkBoxes.add(beginnerTag);
+        checkBoxes.add(easyTag);
+        checkBoxes.add(moderateTag);
+        checkBoxes.add(advancedTag);
+        checkBoxes.add(hardTag);
+        checkBoxes.add(expertTag);
+        checkBoxes.add(sweetTag);
+        checkBoxes.add(sourTag);
+        checkBoxes.add(savoryTag);
+        checkBoxes.add(saltyTag);
+        checkBoxes.add(bitterTag);
+        checkBoxes.add(umamiTag);
+        checkBoxes.add(drinkTag);
+        checkBoxes.add(snackTag);
+        checkBoxes.add(breakfastTag);
+        checkBoxes.add(lunchTag);
+        checkBoxes.add(dinnerTag);
+        checkBoxes.add(dessertTag);
+        checkBoxes.add(beefTag);
+        checkBoxes.add(porkTag);
+        checkBoxes.add(chickenTag);
+        checkBoxes.add(turkeyTag);
+        checkBoxes.add(duckTag);
+        checkBoxes.add(poultryTag);
+        checkBoxes.add(fishTag);
+        checkBoxes.add(lambTag);
+        checkBoxes.add(rabbitTag);
+        checkBoxes.add(shellfishTag);
+        checkBoxes.add(vegetarianTag);
+        checkBoxes.add(veganTag);
+
+
+        createRecipeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                for (CheckBox checkBox : checkBoxes) {
+                    onCheckBoxClicked(checkBox);
+                }
+
+
+                Intent intent = new Intent(addTagsPage.this, NewRecipe.class);
+                intent.putExtra("name", name);
+                intent.putExtra("instructions", getIntent().getStringArrayExtra("instructions"));
+                intent.putExtra("ingredients", getIntent().getStringArrayExtra("ingredients"));
```

Comment:
Could there be a better way of keeping the instructions the user has entered on the page? Maybe instead of creating a new recipe page every time the user goes back from the add tags page, we just don't call `finish` on the new recipe page and somehow set variables of the new recipe page from within the add tags page? Not sure if that's possible, but it'd be better than this method.

Reply:
i will look into that. For now, this is what the internet recommended that I do for now

Reply:
Yeah, this works, and we only have to make it better if we get the time